juju/juju
# Byebye bakery v2 unstable

## Description of change

Update the controller backend to use bakery.v2 instead of bakery.v2-unstable.

There's 3 commits - the first is largely boilerplate.
1. Add the bakery version to client requests which pass in macaroons, including
- login requests
- user registration
- cross model relation calls
(just adding a new attribute to params structs)

2. Update the cross model relations auth backend to user bakery.v2. This includes moving some code from apiserver to live with the rest of the cmr auth code.

3. Update the user auth backend to use bakery.v2

## QA steps

Perform the following tests with both 2.7 and 2.8 clients...

bootstrap a controller
add a user
from a different client, register as that new user
logout and login and ensure juju status works
change user password and check again
login with incorrect password -> error
edit accounts.yaml to alter user name and juju status -> error
delete local cookies and check login

add a new model and create an offer
consume the offer and wait 3 minutes for macaroon to expire
add a cross model relation and check status

bootstrap a 2.7 controller, upgrade, and re-check
